feature_engine/selection/base_recursive_selector.py

```diff
@@ -117,7 +118,7 @@ def __init__(
         self.estimator = estimator
         self.scoring = scoring
         self.threshold = threshold
-        self.cv = cv
+        self.cv = list(cv) if isinstance(cv, GeneratorType) else cv
```

Comment:
Hi @Benetti-Hub thank you so much! This is almost good to go. 

Unfortunately, to remain compatible with sklearn, we can't modify the parameters of init. They need to take the values that the user passes. Could you leave the parameter setting in init as is, an move this line of code to fit?

Reply:
Done! I have moved them

Reply:
Hey @Benetti-Hub 

We are still redefining the self.cv.

What I'd suggest is creating a temporary cv and then passing that onto the cross_validate. I am trying to do that but it does not resolve the issue, so it needs further digging.

The tests also need to check that the output of transform is the desired dataframe, to ensure that the class works as expected and returns the same results when the user passes cv and cv.split(X,y)

Reply:
Hey, given how generators work in Python, it's not possible to create a direct copy of one I am afraid. The issue arises because `self.cv` is used both during the initial fit (which exhausts the generator) and again in the cross-validation, leading to complications.

When I tested a similar scenario in scikit-learn, I observed that the `self.cv` attribute indeed becomes an exhausted generator after the fit call, which modifies it. In fact, you cannot fit two RFECV sequentially using the same generator.

Addressing this problem seems challenging without altering the `super().fit(X, y)` method to include a "cv" argument. Could you let me know if such a modification would be acceptable? If not, I am afraid that the only solution might be to raise a more descriptive error when a generator is passed.

Reply:
oh nvm, i checked the other PR, specifying an additionaly argument also works

Reply:
Hi @Benetti-Hub thank you so much for this message. 

I spend a good hour fighting with the generators and trying to get it to work, and I faced the same issues that you describe in your message. So I am happy to see that I am not the only one, because I was starting to think that I did something wrong and that there is something that I was not seeing through.

Your message gave me a lot of relief.

Anyways, your PR was key for me to get the solution, so I'll credit the work to you. Thank you so much. I'll make a PR during the week for a new release and tag you along the other contributors.

Thanks a lot!

Reply:
Glad I could help!